modules/media_duplicate_validation/src/Plugin/MediaDuplicateValidation/ColorMean.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+namespace Drupal\media_duplicate_validation\Plugin\MediaDuplicateValidation;
+
+use Drupal\media\Entity\Media;
+use Drupal\media\MediaInterface;
+use Drupal\media_duplicate_validation\Plugin\MediaDuplicateValidationBase;
+
+/**
+ * Compares an image to other images.
+ *
+ * To do this we resize all images to a 25x25 grayscale image. Then comparing
+ * the average color value of each row and column we collect a subset of similar
+ * images. Then we compare pixel by pixel and with the thresholds we can mark
+ * each one as similar or not.
+ *
+ * @MediaDuplicateValidation(
+ *   id = "color_mean"
+ * )
+ */
+class ColorMean extends MediaDuplicateValidationBase {
+
+  /**
+   * Database table used for this plugin.
+   */
+  const DATABASE_TABLE = 'media_validate_color_mean';
+
+  /**
+   * Percent different each color can be to be considered the same.
+   */
+  const PIXEL_TOLLERANCE = 25;
+
+  /**
+   * Total percent that is different but still considered similar.
+   */
+  const IMAGE_TOLLERANCE = 10;
+
+  /**
+   * The dimensions to resize the images to compare against.
+   */
+  const RESIZE_DIMENSION = 25;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSimilarItems(MediaInterface $entity) {
+    if (!($file = $this->getFile($entity, ['image']))) {
+      return [];
```

Comment:
Debug log message here? Or that might just create too much noise. I'm indifferent.

Reply:
I think it'd cause too much noise when we have more media entities that dont contain uploaded files. currently even our video entities would cause noise here.